game-core/src/main/java/games/strategy/triplea/ui/ProductionPanel.java

```diff
@@ -198,6 +200,7 @@ protected void calculateLimits() {
 
   protected ResourceCollection getResources() {
     if (bid) {
+      Preconditions.checkNotNull(id, "bid was true while id is null");
```

Comment:
If `id` must be non-`null` when `bid` is `true`, I was going to suggest that be enforced in the constructor.  But this class is a mess.  It looks like both of these fields can be changed at any time.  That makes it very hard to reason about.  I guess throwing an exception at this point is the best anyone can expect.

My only suggestion would be to change this to `checkState()` instead of `checkNotNull()` since we're enforcing a class invariant here.  The message is perfect, however.